File: ios/VectorialData/VectorialData/Core/Models/UserProfile.swift
import Foundation

/// Matches `GET /api/me` response. `convertFromSnakeCase` on JSONDecoder
/// maps `is_subscribed` → `isSubscribed`, etc.
struct UserProfile: Codable, Equatable {
    let email: String
    let isSubscribed: Bool
    let subscriptionStatus: String?
    let deliveryChannel: String?
    let locale: String?
    let createdAt: String?
    let currentPeriodEnd: String?
}

File: android/app/src/main/java/com/vectorialdata/app/core/model/UserProfile.kt
package com.vectorialdata.app.core.model

import kotlinx.serialization.Serializable

/**
 * Matches `GET /api/me`. The snake_case JSON naming strategy maps
 * `is_subscribed` -> [isSubscribed], etc. — mirror of iOS `UserProfile`.
 */
@Serializable
data class UserProfile(
    val email: String,
    val isSubscribed: Boolean = false,
    val subscriptionStatus: String? = null,
    val deliveryChannel: String? = null,
    val locale: String? = null,
    val createdAt: String? = null,
    val currentPeriodEnd: String? = null,
)
